agent/agent_test.go

```diff
@@ -420,6 +420,217 @@ post:
+post:
+  - command: shell.exec
+    params:
+      script: "exit 1"
+`
+	p := &model.Project{}
+	_, err := model.LoadProjectInto(s.ctx, []byte(projYml), nil, "", p)
+	s.NoError(err)
+	s.tc.taskConfig.Project = p
+	s.tc.project = p
+	t := &task.Task{
+		Id:           "task_id",
+		BuildVariant: "some_build_variant",
+		DisplayName:  name,
+		Version:      "my_version",
+	}
+	s.tc.taskModel = t
+	s.tc.taskConfig.Task = t
+	_, err = s.a.runTask(s.ctx, s.tc)
+
+	s.NoError(err)
+	s.Equal("failed", s.mockCommunicator.EndTaskResult.Detail.Status)
+	s.Equal("'shell.exec' (step 1 of 1) in block 'post'", s.mockCommunicator.EndTaskResult.Detail.Description)
+
+	logs := s.getIdleTimeoutLogs()
+	s.Equal("Set idle timeout for 'shell.exec' (step 1 of 1) (test) to 2h0m0s.", logs[0])
+	s.Equal("Set idle timeout for 'shell.exec' (step 1 of 1) in block 'post' (test) to 2h0m0s.", logs[1])
+}
+
+func (s *AgentSuite) TestFailingPostErrorFailsNotTrue() {
+	name := "test_failing_command_after_main_succeeds"
+	projYml := `
+buildvariants:
+- name: some_build_variant
+
+tasks: 
+ - name: test_failing_command_after_main_succeeds
+   commands: 
+    - command: shell.exec
+      params:
+        script: "exit 0"
+post:
+  - command: shell.exec
+    params:
+      script: "exit 1"
+`
+	p := &model.Project{}
+	_, err := model.LoadProjectInto(s.ctx, []byte(projYml), nil, "", p)
+	s.NoError(err)
+	s.tc.taskConfig.Project = p
+	s.tc.project = p
+	t := &task.Task{
+		Id:           "task_id",
+		BuildVariant: "some_build_variant",
+		DisplayName:  name,
+		Version:      "my_version",
+	}
+	s.tc.taskModel = t
+	s.tc.taskConfig.Task = t
+	_, err = s.a.runTask(s.ctx, s.tc)
+
+	s.NoError(err)
+	s.Equal("success", s.mockCommunicator.EndTaskResult.Detail.Status)
+	s.Equal("'shell.exec' (step 1 of 1)", s.mockCommunicator.EndTaskResult.Detail.Description)
+
+	logs := s.getIdleTimeoutLogs()
+	s.Equal(2, len(logs))
+	s.Equal("Set idle timeout for 'shell.exec' (step 1 of 1) (test) to 2h0m0s.", logs[0])
+	s.Equal("Set idle timeout for 'shell.exec' (step 1 of 1) in block 'post' (test) to 2h0m0s.", logs[1])
+}
+
+func (s *AgentSuite) TestSucceedingPostAfterMainSucceeds() {
+	name := "test_failing_command_after_main_succeeds"
+	projYml := `
+buildvariants:
+- name: some_build_variant
+
+post_error_fails_task: true
+tasks: 
+ - name: test_failing_command_after_main_succeeds
+   commands: 
+    - command: shell.exec
+      params:
+        script: "exit 0"
+post:
+  - command: shell.exec
+    params:
+      script: "exit 0"
+`
+	p := &model.Project{}
+	_, err := model.LoadProjectInto(s.ctx, []byte(projYml), nil, "", p)
+	s.NoError(err)
+	s.tc.taskConfig.Project = p
+	s.tc.project = p
+	t := &task.Task{
+		Id:           "task_id",
+		BuildVariant: "some_build_variant",
+		DisplayName:  name,
+		Version:      "my_version",
+	}
+	s.tc.taskModel = t
+	s.tc.taskConfig.Task = t
+	_, err = s.a.runTask(s.ctx, s.tc)
+
+	s.NoError(err)
+	s.Equal("success", s.mockCommunicator.EndTaskResult.Detail.Status)
```

Comment:
In the case of the task succeeding, it seems like we don't need to check the task end description since it's only relevant in the case of failure. I think we can omit this check (and same with the other tests that succeed).

Reply:
done